orchestra/static/orchestra/common/js/helpers.es6.js

```diff
@@ -1,7 +1,30 @@
+const d3 = require('d3')
+
 export default function helpers () {
   return {
     isTaskStaffable: function (taskStatus) {
       return taskStatus !== 'Complete' && taskStatus !== 'Aborted'
+    },
+
+    getSortedTasksSlugs: function (tasks) {
+      return Object.keys(tasks).sort((a, b) => {
+        const previousTask = tasks[a]
+        const nextTask = tasks[b]
+        return d3.ascending(
+          new Date(previousTask.start_datetime),
+          new Date(nextTask.start_datetime)
+        )
+      })
+    },
+
+    getAssigmentsOrderedByList: function (orderedListOfSlugs, assignmentDataList) {
+      return orderedListOfSlugs.reduce((acc, slug) => {
+        const found = assignmentDataList.filter(a => a.stepSlug === slug)
```

Comment:
In the interest of readability, how about something like this?
```js
assignmentDataList.sort(function(a, b){  
  return orderedListOfSlugs.indexOf(a.stepSlug) - orderedListOfSlugs.indexOf(b.stepSlug);
});
```
You might have to change the usage because it sorts the array in-place!